tests/helper_test.py

```diff
@@ -76,3 +97,209 @@ def test_megabyte_numbers_are_converted_with_precision_1(self):
+
+    # TODO: Currently these tests use the ampm format based on whether or not
+    # the testing machine's locale sets them. To be really good mock should be
+    # used to change the locale between an am/pm locale and a 24 hour locale
+    # and test both scenarios.
+
+    __patchers = []
+
+    @classmethod
+    def setUpClass(cls):
+        # Create a random number generator, but seed it so that it will produce
+        # deterministic output. This is used to select a subset of possible
+        # values for each of the tests in this class, since otherwise they
+        # would get really expensive (time wise).
+        cls.random = random.Random()
+        cls.random.seed(42)
+
+        # Pick an exact date to ensure that the tests run the same no matter
+        # what time of day they're run.
+        cls.now = datetime.datetime(2000, 1, 5, 12, 0, 0, 0)
+
+        # Mock datetime.now, which ensures that the time is always the same
+        # removing race conditions from the tests.
+        dt = mock.Mock()
+        dt.now = mock.Mock(return_value=cls.now)
+        cls.__patchers.append(mock.patch('alot.helper.datetime', dt))
+
+        for p in cls.__patchers:
+            p.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        for p in cls.__patchers:
+            p.stop()
+
+    def test_just_now(self):
+        for i in (self.random.randint(0, 60) for _ in xrange(5)):
+            test = self.now - datetime.timedelta(seconds=i)
+            actual = helper.pretty_datetime(test)
+            self.assertEquals(actual, u'just now')
+
+    def test_x_minutes_ago(self):
+        for i in (self.random.randint(60, 3600) for _ in xrange(10)):
+            test = self.now - datetime.timedelta(seconds=i)
+            actual = helper.pretty_datetime(test)
+            self.assertEquals(
+                actual, u'{}min ago'.format((self.now - test).seconds // 60))
+
+    def test_x_hours_ago(self):
+        for i in (self.random.randint(3600, 3600 * 6) for _ in xrange(10)):
+            test = self.now - datetime.timedelta(seconds=i)
+            actual = helper.pretty_datetime(test)
+            self.assertEquals(
+                actual, u'{}h ago'.format((self.now - test).seconds // 3600))
+
+    # TODO: yesterday
+    # TODO: yesterday > now > a year
+    # TODO: last year
+    # XXX: when can the last else be hit?
+
+    @staticmethod
+    def _future_expected(test):
+        if test.strftime('%p'):
+            expected = test.strftime('%I:%M%P').lower()
+        else:
+            expected = test.strftime('%H:%M')
+        expected = expected.decode('utf-8')
+        return expected
+
+    def test_future_seconds(self):
+        test = self.now + datetime.timedelta(seconds=30)
+        actual = helper.pretty_datetime(test)
+        expected = self._future_expected(test)
+        self.assertEqual(actual, expected)
+
+    # Returns 'just now', instead of 'from future or something similar
```

Comment:
the comment (each of these comments) is missing a closing single quote after "future".